lib/sles4sap/console_redirection.pm

```diff
@@ -175,6 +175,11 @@ from serial console. VM ID is collected by opening 'log-console' which is not re
 sub check_serial_redirection {
     # Do not select serial console if it is already done. This avoids log pollution and speeds up process.
     if (is_serial_terminal()) {
+        # Teminate the process when scirpts timed out
+        unless (wait_serial(serial_term_prompt())) {
```

Comment:
Maybe it is not the case here as next command is `select_serial_terminal`, but this code could be problematic. We fall in the same issue and we solved it like here: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20245/files#diff-4b7291e584e1d0b487f685a7b45efc74dcd804ee2236acf7a866003d0f4fb87aR408-R415

Reply:
Yes, it is a good example for check the command line prompt too.